src/Mailgun/Api/AbstractApi.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Mailgun\Api;
+
+use Http\Client\Common\HttpMethodsClient;
+use Mailgun\Exception\HttpServerException;
+use Http\Client\Exception as HttplugException;
+use Mailgun\Serializer\ResponseSerializer;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+abstract class AbstractApi
+{
+    /**
+     * The HTTP client.
+     *
+     * @var HttpMethodsClient
+     */
+    protected $httpClient;
+
+    /**
+     * @var ResponseSerializer
+     */
+    protected $serializer;
+
+    /**
+     * @param HttpMethodsClient  $httpClient
+     * @param ResponseSerializer $serializer
+     */
+    public function __construct(HttpMethodsClient $httpClient, ResponseSerializer $serializer)
```

Comment:
Hm, I would rather use an HttpClient here and wrap the methods client internally (although that would probably require a request factory too, but that's still better IMO). From a configuration point of view accepting HttpClient makes it more flexible.